lib/sensu/server/process.rb

```diff
@@ -572,6 +575,58 @@ def determine_stale_clients
         end
       end
 
+      # Determine stale checks, those that have not sent an event
+      # in a specified amount of time (thresholds). This method
+      # iterates through the result registry. If the time since
+      # the latest event is equal to or greater than a threshold,
+      # the check `:output` is set to a descriptive message, and
+      # `:status` is set to the appropriate non-zero value, and an
+      # alert is sent. If a check has been sending events, then
+      # no alert is published.
+      def determine_stale_checks
+        @logger.info("determining stale checks")
+        @redis.smembers("clients") do |clients|
+          clients.each do |client_name|
+            @redis.keys("result:" + client_name + ":*") do |results|
+              results.each do |result_name|
+                next if result_name.split(':')[2] == 'keepalive'
+                @redis.get(result_name) do |result_json|
+                  next if result_json.nil?
+                  result = MultiJson.load(result_json)
+                  check_defaults = {
+                    :thresholds => {
```

Comment:
I took the defaults from the keepalive staleness check. The reason I think it's important to be opt-out rather than opt-in is because if you change a check config and it's nolonger valid JSON then your check will remain OK and you will have no idea the check isn't being run unless you manually inspect check times.. this is a very bad situation to be in.